plugins/default/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -0,0 +1,694 @@
+                args = EvalParams (setups ++ [section]) _textDocument
+                cmd' =
+                  (cmd :: Command)
+                    { _arguments = Just (List [toJSON args]),
+                      _title =
+                        if trivial resultRange
+                          then "Evaluate..."
+                          else "Refresh..."
+                    }
+        ]
+
+  dbg "SETUPS" setups
+
+  dbg_ $
+    unwords
+      [ show (length tests),
+        "tests in",
+        show (length nonSetups),
+        "sections",
+        show (length setups),
+        "setups",
+        show (length lenses),
+        "lenses."
+      ]
+
+  now' <- liftIO getCurrentTime
+  dbg "TIME" (diffUTCTime now' now)
+
+  -- dbg "LENSES" lenses
+  return $ List lenses
+  where
+    trivial (Range p p') = p == p'
+
+evalCommandName :: CommandId
+evalCommandName = "evalCommand"
+
+evalCommand :: PluginCommand
+evalCommand = PluginCommand evalCommandName "evaluate" runEvalCmd
+
+-- |Specify test section to execute
+data EvalParams = EvalParams
+  { sections :: [Section],
+    module_  :: !TextDocumentIdentifier
+  }
+  deriving (Eq, Show, Generic, FromJSON, ToJSON)
+
+runEvalCmd :: CommandFunction EvalParams
+runEvalCmd lsp state EvalParams {..} = withIndefiniteProgress lsp "Evaluating" Cancellable $ response' $ do
+  now <- liftIO getCurrentTime
+
+  let TextDocumentIdentifier {_uri} = module_
+  fp <- handleMaybe "uri" $ uriToFilePath' _uri
+  let nfp = toNormalizedFilePath' fp
+  contents <- liftIO $ getVirtualFileFunc lsp $ toNormalizedUri _uri
+  text <- handleMaybe "contents" $ virtualFileText <$> contents
+
+  session  <-
+    liftIO $
+      runAction "runEvalCmd.ghcSessionDeps" state $
+        use_ GhcSessionDeps nfp
+
+  ms <-
+    liftIO $
+      runAction "runEvalCmd.getModSummary" state $
+        use_ GetModSummary nfp
+
+  let tests = testsBySection sections
+
+  let modName = moduleName $ ms_mod ms
+
+  let Just impPath = moduleImportPath nfp modName
+  liftIO $ dbg "IMPORT PATH" impPath
+
+  -- Setup environment for evaluation
+  hscEnv' <- withSystemTempFile (takeFileName fp) $ \logFilename logHandle -> ExceptT $ (either Left id <$>) . gStrictTry $
+  -- Right hscEnv' <- liftIO $ withSystemTempFile (takeFileName fp) $ \logFilename logHandle ->  (either Left id <$>) . gStrictTry $
+      evalGhcEnv (hscEnv session) $ do
+        env <- getSession
+        dbg "ENV SESSION DFLAGS".  ic_dflags . hsc_IC  $ env
+
+        -- Get options and language flags from module source
+        df0 <- liftIO $ setupDynFlagsForGHCiLike env $ ms_hspp_opts ms
+        _ <- setSessionDynFlags df0
+        dbg "df0 imports" $ importPaths df0
+
+        -- We add back the local importPath, so that we can find local dependencies
+        --let df = df0 {importPaths=[impPath]}
+        --_lp <- setSessionDynFlags df
+        df <- modifyFlags (\df -> df {importPaths=[impPath]})
+        dbg "df imports" $ importPaths df0
+        --dbg "df" df
```

Comment:
```suggestion
      evalGhcEnv (hscEnvWithImportPaths session) $ do
        env <- getSession
        dbg "ENV SESSION DFLAGS".  ic_dflags . hsc_IC  $ env

        -- Get options and language flags from module source
        df <- liftIO $ setupDynFlagsForGHCiLike env $ ms_hspp_opts ms
        _ <- setSessionDynFlags df0
        dbg "df imports" $ importPaths df
```

Reply:
hscEnvWithImportPaths !

So much to lear :-)